src/views/PokemonInfoView.tsx

```diff
@@ -0,0 +1,128 @@
+"use client";
+import { getAbility, getPokemonById, getRandomPokemonId } from "@/utils/pokemon";
+import { useQueryParams } from "@/hook/useQueryParams";
+import { PokemonSearchBar } from "@/components/PokemonSearchBar";
+import PokemonInfoCard from "@/components/PokemonInfoCard";
+import { useEffect, useState } from "react";
+import { Pokemon } from "pokenode-ts";
+import { capitalizeFirstLetter } from "@/utils/functional";
+
+export default function PokemonInfoView() {
+    const { queryParams, setQueryParams } = useQueryParams<{
+        pokemon: string;
+      }>();
+
+    const [pokemonId, setPokemonId] = useState(
+    queryParams.pokemon
+        ? parseInt(queryParams.pokemon)
+        : getRandomPokemonId()
+    );
+
+    const [pokemon, setPokemon] = useState<Pokemon | null>(null);
+    const [stats, setStats] = useState(new Map<string, any>());
+    const [abilities, setAbilities] = useState(new Map<string, string>());
```

Comment:
it wasn't happy being null, so I had to make it an empty map

Reply:
I think it's better to check for `null` and have an explicit fall back value `abilities?.get(<some key>) ?? <some fallback value>` rather than just calling `.get` because that `get` can return `undefined` meaning it can crashes easily if we're not careful